whois-commons/src/main/resources/nrtm_schema.sql

```diff
@@ -19,58 +35,72 @@ create table `source`
-    UNIQUE KEY `version_information__type__last_serial_id_uk` (`type`, `last_serial_id`)
+    CONSTRAINT `version__source__fk` FOREIGN KEY (`source_id`) REFERENCES `source` (`id`),
+    UNIQUE KEY `version__source__version__uk` (`source_id`, `version`),
+    UNIQUE KEY `version__session__version__uk` (`session_id`, `version`),
+    UNIQUE KEY `version__type__last_serial_id__uk` (`type`, `last_serial_id`)
 ) ENGINE = InnoDB
   DEFAULT CHARSET = utf8;
 
-DROP TABLE IF EXISTS `snapshot_file`;
-create table `snapshot_file`
+CREATE TABLE `snapshot_object`
 (
-    `id`         int(10) unsigned NOT NULL AUTO_INCREMENT,
-    `version_id` int(10) unsigned NOT NULL DEFAULT '0',
-    `name`       varchar(256)     NOT NULL DEFAULT '',
-    `hash`       varchar(128)     NOT NULL DEFAULT '',
-    `created`    bigint unsigned  not null default unix_timestamp(),
+    `id`          int unsigned NOT NULL AUTO_INCREMENT,
+    `version_id`  int unsigned NOT NULL,
+    `object_id`   int unsigned NOT NULL,
+    `sequence_id` int unsigned NOT NULL,
+    `rpsl`        longtext     NOT NULL,
     PRIMARY KEY (`id`),
-    CONSTRAINT `snapshot_file__version_information_fk` FOREIGN KEY (`version_id`) REFERENCES `version_information` (`id`)
+    UNIQUE KEY `snapshot_object__object_id__uk` (`object_id`),
+    CONSTRAINT `snapshot_object__version_id__fk` FOREIGN KEY (`version_id`) REFERENCES `version_info` (`id`)
 ) ENGINE = InnoDB
   DEFAULT CHARSET = utf8;
 
-DROP TABLE IF EXISTS `delta_file`;
-create table `delta_file`
+CREATE TABLE `snapshot_file`
 (
     `id`         int unsigned    NOT NULL AUTO_INCREMENT,
-    `version_id` int unsigned    NOT NULL DEFAULT '0',
-    `name`       varchar(256)    NOT NULL DEFAULT '',
-    `payload`    longblob        NOT NULL DEFAULT '',
-    `hash`       varchar(128)    NOT NULL DEFAULT '',
-    `created`    bigint unsigned not null default unix_timestamp(),
+    `version_id` int unsigned    NOT NULL,
+    `name`       varchar(256)    NOT NULL,
+    `hash`       varchar(256)    NOT NULL,
+    `created`    bigint unsigned NOT NULL,
     PRIMARY KEY (`id`),
-    CONSTRAINT `delta_file__version_information_fk` FOREIGN KEY (`version_id`) REFERENCES `version_information` (`id`)
+    UNIQUE KEY `snapshot_file__version_id__uk` (`version_id`),
+    UNIQUE KEY `snapshot_file__name__uk` (`name`),
+    CONSTRAINT `snapshot_file__version_id__fk` FOREIGN KEY (`version_id`) REFERENCES `version_info` (`id`)
 ) ENGINE = InnoDB
   DEFAULT CHARSET = utf8;
 
-DROP TABLE IF EXISTS `snapshot_object`;
-create table `snapshot_object`
+CREATE TABLE `delta_file`
 (
-    `id`         int unsigned NOT NULL AUTO_INCREMENT,
-    `whois_pkey` varchar(256) NOT NULL DEFAULT '',
-    `payload`    longblob     NOT NULL DEFAULT '',
+    `id`         int unsigned    NOT NULL AUTO_INCREMENT,
+    `version_id` int unsigned    NOT NULL,
+    `name`       varchar(256)    NOT NULL,
+    `hash`       varchar(256)    NOT NULL,
+    `payload`    longtext        NOT NULL,
```

Comment:
Is payload the content of the delta file? Is it necessary to store separately in the database? We could instead refer to a range in `snapshot_object`